plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +359,23 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*MaxPriority.
+		if checkIntegerOverFlow(totalPriority, config.Weight*MaxPriority) {
+			return nil, fmt.Errorf("Total priority of priority functions has overflown")
+		}
+		totalPriority += config.Weight * MaxPriority
+	}
 	return configs, nil
 }
 
+// checkIntegerOverFlow checks if totalPriority can overflow after adding it with weight.
+func checkIntegerOverFlow(totalPriority int, weight int) bool {
```

Comment:
I was just suggesting that the entire `addPriority...` operation should handle the TODO as well as the other stuff all in one place, and return a new value representing the result of the operation.  That makes it really easy to read, unit test, and so on.